primo/utils/clustering_utils.py

```diff
@@ -97,39 +101,67 @@ def distance_matrix(wd: WellData, weights: dict) -> np.ndarray:
     )
 
 
-def perform_clustering(wd: WellData, distance_threshold: float = 10.0):
+def _well_clusters(wd: WellData) -> dict:
+    """Returns well clusters"""
+    col_names = wd.col_names
+    set_clusters = set(wd[col_names.cluster])
+    return {
+        cluster: wd.data[wd[col_names.cluster] == cluster].index.to_list()
+        for cluster in set_clusters
+    }
+
+
+def _check_existing_cluster(wd: WellData):
     """
-    Partitions the data into smaller clusters.
+    Checks if clustering has already been performed on the WellData object.
 
     Parameters
     ----------
-    distance_threshold : float, default = 10.0
-        Threshold distance for breaking clusters
+    wd : WellData
+        The WellData object to check for existing clusters.
 
     Returns
     -------
-    n_clusters : int
-        Returns number of clusters
+    int or None
+        Number of clusters if clustering exists, otherwise None.
     """
     if hasattr(wd.col_names, "cluster"):
-        # Clustering has already been performed, so return.
-        # Return number of cluster.
+        # Clustering has already been performed, so return the number of clusters.
         LOGGER.warning(
-            "Found cluster attribute in the WellDataColumnNames object."
+            "Found cluster attribute in the WellDataColumnNames object. "
             "Assuming that the data is already clustered. If the corresponding "
             "column does not correspond to clustering information, please use a "
             "different name for the attribute cluster while instantiating the "
             "WellDataColumnNames object."
         )
         return len(set(wd[wd.col_names.cluster]))
+    return None
```

Comment:
It is a good idea in general to pass the same return type from a function on all branches. In this case, we are passing number of clusters in one, and None in another. Perhaps -1 is a better return to check for no clustering? I would have set this as a boolean (True/False) since the method name is "check_existing_cluster"

Reply:
I like the True/False approach. If you need the number of clusters somewhere else, it's one line of code once you have checked if the attribute exists or not.

Reply:
Changed the returns to True/False